src/Native/Unix/System.Native/pal_time.c

```diff
@@ -167,22 +167,22 @@ int32_t SystemNative_GetCpuUtilization(ProcessCpuInformation* previousCpuInfo)
     }
 
     uint64_t resolution = SystemNative_GetTimestampResolution();
-    uint64_t timestamp = SystemNative_GetTimestamp();
+    double timestamp = SystemNative_GetTimestamp();
 
-    uint64_t currentTime = timestamp * SecondsToNanoSeconds / resolution;
```

Comment:
this should probably just be `uint64_t currentTime = (uint64_t)(timestamp * ((double)SecondsToNanoSeconds / resolution));`

Reply:
and everything else should stay as `uint64_t`. That will help minimize overall error and most closely matches what full framework is doing.